android/app/build.gradle

```diff
@@ -21,6 +21,10 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
+def googleMapsDevApiKey = localProperties.getProperty('googleMaps.apiKeyDev') ?: System.getenv("BITRISEIO_ANDROID_MAPS_KEY_DEV") ?: ""
+def googleMapsProdApiKey = localProperties.getProperty('googleMaps.apiKeyProd') ?: System.getenv("BITRISEIO_ANDROID_MAPS_KEY_PROD") ?: ""
```

Comment:
w jaki sposób w tym podejściu podajesz klucz API budując apkę lokalnie? bezpośrednio w `manifestPlaceholder` wklejasz? też widzę, że na ios nie ma obsługi podawania różnych kluczy api w zależności od środowiska. Klucz do API NyTimes trzymamy w pliku, dlaczego ten klucz api jest obsługiwany w inny sposób? Jak widać sporo wad ma to rozwiązanie. 

Proponuję uspójnić trzymanie kluczy poprzez trzymanie ich wszystkich w jednym pliku z kluczami, może to być `secrets.json`, który mamy obecnie. Przekazywanie kluczy do programu w CI zrobiłbym poprzez `--dart-define` i wyciągał już w samym darcie, żeby ominąć podział na platformy ([tu artykuł o tym](https://itnext.io/flutter-1-17-no-more-flavors-no-more-ios-schemas-command-argument-that-solves-everything-8b145ed4285d), dotyczy on flavorów, ale w tym przypadku też się nada). Co myślisz?

Reply:
Nie wklejam do manifestPlaceholder. Przy obecnym rozwiązaniu w kwestii androida należy dodać zmienne
```
googleMaps.apiKeyProd=XYZ
googleMaps.apiKeyDev=XYZ
``` w pliku `local.properties`, jeżeli są takowe lokalnie to zaciąga lokalnie, jak nie ma lokalnie to bierze ze zmiennej środowiskowej. iOSa pod flavory chciałem wyrzucić jako osobny task, bo z wiadomych przyczyn nie miałem jak tego weryfikować i w sumie skupiłem się na androidzie. W kontekście samego androida nie powiedziałbym, że kolejne klucze/środowiska nie stanowią jakiegoś problemu przy tym rozwiązaniu, chyba że nie czegoś nie widzę? 🤔 Swoją drogą api key jest dorzucany do wersji produkcyjnej na bitrise jakoś? Bo chciałem sprawdzić ale ściągnąłem kilka wersji i w sumie każda się wywala po splash screenie 🤔 Wracając do rozwiązania to jutro przejrzę artykuł i się wypowiem ;)

Reply:
>Swoją drogą api key jest dorzucany do wersji produkcyjnej na bitrise jakoś?

Nie jest :D przegapiliśmy to kompletnie, apka która buduje się poprzez bitrise nie będzie w ogóle działać :p tym bardziej myślę, że warto to uspójnić. Jeżeli wrzucalibyśmy te klucze poprzez `--dart-define` to też nie trzeba by było rozróżniać ich na bitrise ze względu na środowisko, po prostu dany workflow wrzucałby odpowiedni klucz, tak samo możemy wrzucać API key do samego NYTimesa, jeżeli wolisz to możemy zrobić to wszystko jako osobny task

Reply:
Przeczytałem ocb i fajne rozwiązanie. Faktycznie mogłem od razu wrzucić do auth/secrets.json bo zasada działania identyczna. No nie wiedziałem jak to w iosie funkcjonuje i że nie da się tak. Zaraz wrzucę poprawki. Klucz do api wrzuciłem poprzez dart-define i wyciągnąłem z w darcie, ale już z kluczami do google maps tak nie zrobię, bo one muszą być zadeklarowane w natywnej części apki. Androida spoko zrobiłem, fajnie działa, ale no iOSa nie jestem w stanie, więc osobny task i jak będziesz miał ochotę to śmiało, a jak nie to może kiedyś ogarnę maca :D

Reply:
Ok, z racji, że zajmowałeś się mapami dodaj proszę taska na ogarnięcie map na iOS do trello :)